src/js/smooch.jsx

```diff
@@ -196,6 +195,40 @@ export class Smooch {
         });
     }
 
+    monitorUrlChanges() {
+        const onhashchange = window.onhashchange;
+        const pushState = window.history.pushState;
+        const replaceState = window.history.replaceState;
+
+        window.onhashchange = function() {
+            this.updateNowViewing();
+            onhashchange && onhashchange.apply(window, arguments);
+        }.bind(this);
+
+        window.history.pushState = function(state, title, url) {
+            if (url) {
+                this.updateNowViewing();
+            }
+            pushState && pushState.apply(window.history, arguments);
+        }.bind(this);
+
+        window.history.replaceState = function(state, title, url) {
+            if (url) {
+                this.updateNowViewing();
+            }
+            replaceState && replaceState.apply(window.history, arguments);
+        }.bind(this);
+    }
+
+    updateNowViewing() {
```

Comment:
as discussed, you can merge the logic of getting the device id in the service and just hide the implementation in the service. The service holds the business logic.